client/scss/components/_listing.scss

```diff
@@ -223,6 +223,10 @@ ul.listing {
 
   .title {
     word-break: break-word;
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    gap: theme('spacing.2');
```

Comment:
Unfortunately this caused unexpected styling issues elsewhere 🤦 
![image](https://github.com/wagtail/wagtail/assets/6379424/59c369ee-664f-4b28-b426-c566a5387c51)

I'll create a PR to fix this